shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/Alert.kt

```diff
@@ -203,4 +205,64 @@ data class Alert(
     fun anyInformedEntity(predicate: (InformedEntity) -> Boolean) = informedEntity.any(predicate)
 
     fun matchingEntities(predicate: (InformedEntity) -> Boolean) = informedEntity.filter(predicate)
+
+    companion object {
+        /**
+         * Filter the list of alerts to only the ones that match the given filter parameters and
+         * affect boarding.
+         */
+        fun filter(
+            alerts: List<Alert>,
+            directionId: Int?,
+            routeIds: List<String>,
+            stopIds: Set<String>
+        ): List<Alert> {
+            return applicableAlerts(
+                routes = routeIds,
+                stopIds = stopIds,
+                directionId = directionId,
+                alerts = alerts
+            )
+        }
+
+        /**
+         * Gets the alerts of the first stop that is downstream of the target stop which has alerts.
+         *
+         * @param alerts: The full list of alerts
+         * @param trip: The trip used to calculate downstream stops
+         * @param targetStopWithChildren: The child and parent stop Ids of the target stop
+         */
+        fun downstreamAlerts(
+            alerts: Collection<Alert>,
+            trip: Trip,
+            targetStopWithChildren: Set<String>,
+        ): List<Alert> {
+            val indexOfTargetStopInPattern =
+                trip.stopIds?.indexOfFirst { targetStopWithChildren.contains(it) }
+            if (
+                indexOfTargetStopInPattern != null &&
```

Comment:
🎱  (8ball emoji means optional/stylistic, I have no idea why.) The `?:` operator could probably clean up this boolean and make it IMO more readable.

Reply:
When assigning the variables, I mean. E.g. `val stopIdList = trip.stopIds ?: emptyList()` would remove a few null checks.

Reply:
I like that!